backend/application/api/src/main/kotlin/io/raemian/api/emoji/controller/response/ReactedEmojisResponse.kt

```diff
@@ -5,61 +5,79 @@ import io.raemian.storage.db.core.user.User
 
 data class ReactedEmojisResponse(
     val totalReactedEmojisCount: Int,
-    val reactedEmojis: Map<Long, ReactedEmojiDTO>,
+    val latestReactUserNickname: String?,
+    val reactedEmojis: List<ReactedEmojiAndReactUsers>,
 ) {
     companion object {
-        fun of(reactedEmojis: List<ReactedEmoji>): ReactedEmojisResponse {
-            val mapValues = reactedEmojis
-                .filter { it.emoji.id != null }
-                .groupBy { it.emoji.id!! }
-                .mapValues(ReactedEmojiDTO.Companion::from)
+        fun of(reactedEmojis: List<ReactedEmoji>, username: String): ReactedEmojisResponse {
+            val reactedEmojiAndReactUsers = convert(reactedEmojis, username)
 
-            val totalEmojisCount = mapValues.values.sumOf { it.count }
-            return ReactedEmojisResponse(totalEmojisCount, mapValues)
+            val totalEmojisCount = reactedEmojiAndReactUsers.sumOf { it.reactCount }
+            val latestReactUserNickname = reactedEmojis.lastOrNull()?.reactUser?.nickname
+            return ReactedEmojisResponse(
+                totalReactedEmojisCount = totalEmojisCount,
+                latestReactUserNickname = latestReactUserNickname,
+                reactedEmojis = reactedEmojiAndReactUsers,
+            )
         }
+
+        private fun convert(
+            reactedEmojis: List<ReactedEmoji>,
+            username: String,
+        ): List<ReactedEmojiAndReactUsers> =
+            reactedEmojis
+                .filter { it.emoji.id != null }
+                .groupBy { it.emoji.id!! }
```

Comment:
```suggestion
                .groupBy { it.emoji.id }
```

filter 에서 널체크 하면
groupBy 에서 `!!`로 강제할당 안해줘도 되야할거같은데
안되나보네?!?!?!?
그냥 갑자기 보여서 ㅎㅎ...